File: DSA/Python/Leetcode-Array/Easy/Longest_common_prefix.py
def Longest_common_prefix_bruteforce(strs):
    
    prefix = ""

    for i in range(len(strs[0])):

        for j in range(1 , len(strs)):
            
            if i > len(strs[j]) or strs[j][i] != strs[0][i]:
                return prefix
            
        prefix += strs[0][i]
    return prefix




print(Longest_common_prefix_bruteforce(["flower" , "flow" , "flight"]))


def longest_common_prefix(strs):
    if not strs:
        print("Empty")
        return

    prefix = strs[0]

    for i in range(1, len(strs)):
        j = 0
        print(j)  

        while j < len(prefix) and j < len(strs[i]) and prefix[j] == strs[i][j]:
            j += 1

        new_prefix = ""
        print(j)  

        for k in range(j):
            new_prefix += prefix[k]

        prefix = new_prefix

    print(prefix)
longest_common_prefix(["flower" , "flow" , "flight"])

File: DSA/C++/Leetcode-Array/Easy/Longest_Common_Prefix.cpp
#include <iostream>
#include <vector>
#include <string>








void LongestCommonPrefix(std::vector<std::string>& strs){

    if (strs.empty()) std:: cout << "Enmty";


    std::string prefix  = strs[0];
   
    for (int i = 1 ; i < strs.size(); i++){
        int j = 0;

        std::cout << j << std::endl;
        
        while (j < prefix.size() && j < strs[i].size() && prefix[j] == strs[i][j]){
            j++;
        }
        
       std::string newPrefix = "";
       std::cout << j << std::endl;
       for(int k = 0 ; k < j; k++){
        
        newPrefix += prefix[k];
       }

       
       prefix = newPrefix;

    }

    std::cout << prefix;


}



int main(){


  std::vector<std::string> strs = {"flower" , "flow" , "flight"};
  LongestCommonPrefix(strs);

}